### PR TITLE
Add MLContextMLogregTest

### PR DESCRIPTION
**Pull request by omkarghugarkar007**
1. Tried out by removing the comments of the Lines and made changes to the code.
2. Currently getting the following error 
image

3. Debugging the code to get rid of the error.

### CODE REVIEW

File: src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextMLogregTest.java [hunk 1]
```diff
@@ -60,30 +60,30 @@ public void testMLogregDense() {
...
 		fullRScriptName = "src/test/scripts/functions/codegenalg/Algorithm_MLogreg.R";
 
-//		rCmd = getRCmd(inputDir(), String.valueOf(alpha),
-//				String.valueOf(maxiter), expectedDir());
-//		runRScript(true);
+		writeInputMatrixWithMTD("X", X, true);
+		writeInputMatrixWithMTD("Y", y, true);
+
+		rCmd = getRCmd(inputDir(), String.valueOf(2),String.valueOf(epsilon),String.valueOf(maxiter), expectedDir());
+		runRScript(true);
```
Comment (j143) [inline src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextMLogregTest.java:76]: **comment 1**
Do not run `R` script yet, first of all make sure the `dml` runs fine.

File: src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextMLogregTest.java [hunk 2]
```diff
@@ -60,30 +60,30 @@ public void testMLogregDense() {
...
-		MLResults outmat = ml.execute(mlr);//.getMatrix("w").toMatrixBlock();
+		mlr.in("X", X).in("Y", y).in("icpt", 2).in("tol", epsilon).in("moi", maxiter).in("reg", 0.001).out("w");
+		outmat = ml.execute(mlr).getMatrix("w").toMatrixBlock();
 
 
 		//compare matrices
-//		HashMap<MatrixValue.CellIndex, Double> rfile = readRMatrixFromFS("w");
-//		TestUtils.compareMatrices(rfile, outmat, eps);
+		HashMap<MatrixValue.CellIndex, Double> rfile = readRMatrixFromFS("w");
+		TestUtils.compareMatrices(rfile, outmat, eps);
```
Comment (j143) [inline src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextMLogregTest.java:87]: **comment 3**
Compare the results after the `R` funtions!

File: src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextMLogregTest.java [hunk 3]
```diff
@@ -60,30 +60,30 @@ public void testMLogregDense() {
...
 
-		baseDirectory = "target/testTemp/functions/mlcontext/";
 
 		//generate actual datasets
 		double[][] X = getRandomMatrix(rows, cols, 0, 1, sparse?sparsity2:sparsity1, 2384);
-		writeInputMatrixWithMTD("X", X, true);
 		double[][] y = TestUtils.round(getRandomMatrix(rows, 1, 0.51, 5+0.49, 1.0, 9283));
-		writeInputMatrixWithMTD("Y", y, true);
+
+		baseDirectory = "target/testTemp/functions/mlcontext/";
```
Comment (j143) [inline src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextMLogregTest.java:68]: **comment 2**
- Based on the `baseDirectory`, the `fullDMLScriptName` will be influenced.
- and input/output data will be available at `baseDirectory/in/`, `baseDirectory/out/` respectively.

**Review by j143 — CHANGES_REQUESTED** [review]

**Comment by omkarghugarkar007** [comment]
Issue Solved.
1. Need to make a small change in the DML script.
2. Name of the Output variable of DMLscript and java output variable was different, thus was getting an error.

**Comment by j143** [comment]
Thank you, @omkarghugarkar007 . I will merge this after a deep pass.

---
As for the next task, have a look at https://github.com/j143/systemds/commits/GLM_MLContext 

- We will work on `GLM`
- This script is a bit tricky and long (try your best!)

**Pull request closed without merging**